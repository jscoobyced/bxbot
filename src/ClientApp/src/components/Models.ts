export class Constants {
    public static readonly StaticContent = (process.env.mode === 'development') ? '' : '/dist';

    public static readonly ImagePath = `${Constants.StaticContent}/images/`;
}

export interface SelectOption {
    value: number;
    text: string;
}
