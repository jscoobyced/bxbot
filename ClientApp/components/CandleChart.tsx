import * as React from 'react';
import { RouteComponentProps } from 'react-router';
import { GoogleCharts } from 'google-charts';
import { DateUtil } from '../utils/date';

interface FetchDataState {
    pairings: Pairing[];
    loading: boolean;
}

export class CandleChart extends React.Component<RouteComponentProps<{}>, FetchDataState> {
    private static ChartData: Array<Pairing>;

    constructor() {
        super();
        this.state = { pairings: [], loading: true };

        fetch('api/Data/pairing/1/5')
            .then(response => response.json() as Promise<Pairing[]>)
            .then(data => {
                CandleChart.ChartData = data;
                this.setState({ loading: false });
            });
    }

    public render() {
        let contents: JSX.Element = <span></span>;
        if (this.state.loading) {
            contents = <p><em>Loading...</em></p>
        }
        else {
            GoogleCharts.load(this.drawChart);
        }

        return <div>
            <h1>BTC on bx.in.th</h1>
            <div id="chart_div"></div>
            {contents}
        </div>;
    }

    private drawChart() {
        const rawData: Array<Pairing> = CandleChart.ChartData;
        const formatedData = new Array<any>();
        const lengthToShow = 96;
        const startIndex = Math.max(0, rawData.length - lengthToShow);
        for (let i = startIndex; i < rawData.length; i++) {
            formatedData.push([
                DateUtil.toHumanHours(new Date(rawData[i].timestamp - 7 * 60 * 60 * 1000)),
                rawData[i].low,
                rawData[i].open,
                rawData[i].close,
                rawData[i].high
            ]);
        }
        var data = GoogleCharts.api.visualization.arrayToDataTable(formatedData, true);

        var options = {
            legend: 'none',
            height: 500,
            candlestick: {
                fallingColor: { strokeWidth: 0, fill: '#a52714' },
                risingColor: { strokeWidth: 0, fill: '#0f9d58' },
                hollowIsRising: true
            },
            colors: ['black']
        };

        var chart = new GoogleCharts.api.visualization.CandlestickChart(document.getElementById('chart_div'));

        chart.draw(data, options);
    }
}

interface Pairing {
    timestamp: number;
    low: number;
    high: number;
    close: number;
    open: number;
}