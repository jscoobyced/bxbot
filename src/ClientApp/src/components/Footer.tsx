import * as React from 'react';

const version = process.env.VERSION;

export class Footer extends React.Component {
    public render() {
        return <footer>${version}</footer>;
    }
}
