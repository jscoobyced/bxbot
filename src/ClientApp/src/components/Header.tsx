import * as React from 'react';
import { NavLink } from 'react-router-dom';
import { ContentKey } from '../services/ContentKey';
import { ContentService } from '../services/ContentService';

export class Header extends React.Component {

    public render() {
        const websiteName = ContentService.getContent(ContentKey.WebsiteName);
        return <header>
            <img src='/images/wings.svg' className='logo' alt={websiteName + ' logo'} title={websiteName} />
            <span>{websiteName}</span>
            <nav>
                <NavLink to={'/'}>
                    Home
                </NavLink>
                <a href='#' className='menu-space'>&nbsp;</a>
                <a href='#' className='dropdown'>Features</a>
                <div className="dropdown-content">
                <NavLink to={'/candle'}>
                        CandleChart
                    </NavLink>
                    <NavLink to={'/'}>
                        Stuff #1
                    </NavLink>
                    <NavLink to={'/candle'}>
                        Stuff #2
                    </NavLink>
                </div>
                <NavLink to={'/'}>
                    About
                </NavLink>
            </nav>
        </header>;
    }
}