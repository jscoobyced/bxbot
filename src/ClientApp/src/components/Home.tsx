import * as React from 'react';

export class Home extends React.Component {
    public render() {
        return <article>
            <h1>Hello, world!</h1>
        </article>;
    }
}
