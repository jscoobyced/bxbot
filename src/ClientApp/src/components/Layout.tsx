import * as React from 'react';
import { Link, NavLink } from 'react-router-dom';

export interface LayoutProps {
    children?: React.ReactNode;
}

export class Layout extends React.Component<LayoutProps, {}> {
    public render() {
        return <div>
            <div>
                <NavLink to={'/'}>
                    Home
                </NavLink> |&nbsp;
                <NavLink to={'/candle'}>
                    CandleChart
                </NavLink>
            </div>
            {this.props.children}
        </div>;
    }
}
