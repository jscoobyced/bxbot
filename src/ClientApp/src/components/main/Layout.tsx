import * as React from 'react';
import { AllContent } from '../../services/Models';
import { Footer } from './Footer';
import { Header } from './Header';

export interface LayoutProps {
    children?: React.ReactNode;
    allContent?: AllContent;
}

export class Layout extends React.Component<LayoutProps> {

    public constructor(props: LayoutProps) {
        super(props);
    }

    public render() {
        if (this.props.allContent == null) {
            return <div></div>;
        }

        return <div className='content'>
            <Header navigationMenuContent={this.props.allContent.navigationMenuContent} />
            <main>
                {this.props.children}
            </main>
            <Footer />
        </div>;
    }
}
