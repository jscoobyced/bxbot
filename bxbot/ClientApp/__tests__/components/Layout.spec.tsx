import * as React from 'react';
import { shallow, mount, render } from 'enzyme';
import { Layout } from '../../components/Layout';
import { } from 'jest';
import * as ShallowRenderer from 'react-test-renderer/shallow';

describe('Layout component', () => {
    it('should match snapshot', () => {
        const renderer = ShallowRenderer.createRenderer()
        const layout = renderer
            .render(<Layout />);
        expect(layout).toMatchSnapshot();
    })
});