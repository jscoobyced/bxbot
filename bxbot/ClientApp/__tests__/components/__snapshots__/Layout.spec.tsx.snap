// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout component should match snapshot 1`] = `
<div
  className="container-fluid"
>
  <div
    className="row"
  >
    <div
      className="col-sm-1"
    >
      <NavMenu />
    </div>
    <div
      className="col-sm-11"
    />
  </div>
</div>
`;
